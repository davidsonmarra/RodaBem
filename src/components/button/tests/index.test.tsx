import React from 'react';
import {create} from 'react-test-renderer';
import Button, {ButtonType} from '..';

const mockText = 'Snapshot test!';
const createComponent = ({
  text = mockText,
  type = ButtonType.primary,
  ...rest
}) =>
  create(
    <Button text={text} type={type} {...rest}>
      {mockText}
    </Button>,
  );

describe('Button', () => {
  it('renders correctly', () => {
    const tree = createComponent({}).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders children correctly', () => {
    const instance = createComponent({}).root;
    expect(instance.props.children).toBe(mockText);
  });

  it('should call onPress', () => {
    const onPress = jest.fn();
    const instance = createComponent({onPress}).root;
    instance.props.onPress();
    expect(onPress).toHaveBeenCalled();
  });

  it('renders primary style correctly', () => {
    const instance = createComponent({}).root;
    expect(instance.props.type).toEqual(ButtonType.primary);
  });

  it('renders secondary style correctly', () => {
    const instance = createComponent({type: ButtonType.secondary}).root;
    expect(instance.props.type).toEqual(ButtonType.secondary);
  });
});
