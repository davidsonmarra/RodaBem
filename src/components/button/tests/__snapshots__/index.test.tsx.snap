// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "red",
      "borderRadius": 12,
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingHorizontal": 8,
      "paddingVertical": 12,
      "width": "100%",
    }
  }
>
  <Text
    style={
      [
        {
          "borderRadius": 4,
          "color": "white",
          "fontSize": 16,
          "fontWeight": "bold",
          "padding": 8,
        },
        undefined,
      ]
    }
  >
    Snapshot test!
  </Text>
</View>
`;
